### PR TITLE
Adds simple script for profiling NonhydrostaticModel

### PR DESCRIPTION
**Pull request by glwagner**

### CODE REVIEW

**Comment by hennyg888** [comment]
Thank you @glwagner ! This is much better than the very bare-bones script I was profiling.

File: benchmark/profiliable_nonhydrostatic_simulation.jl [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+@info "warming up"
+
+run!(simulation)
+
+simulation.stop_iteration += 10
+
+if model.architecture isa GPU
+    CUDA.@profile run!(simulation)
+else
+    @profile run!(simulation)
```
Comment (hennyg888) [inline benchmark/profiliable_nonhydrostatic_simulation.jl:42]: Outputs and formats profiling results into a sorted flat (not tree) text table. Saves into text file. Could also put in `mincount=100` to shorten outputs by removing negligible results.
```suggestion
    Profile.clear()
    @profile run!(simulation)
    open("nonhydrostatic_profile.txt", "w") do io
        Profile.print(IOContext(io, :displaysize => (1000, 350)), format=:flat, sortedby=:count)
    end
```

Comment (glwagner) [reply benchmark/profiliable_nonhydrostatic_simulation.jl:42]: What about using `StatProfilerHTML`? I got some mileage out of this on #1923 .

Comment (glwagner) [reply benchmark/profiliable_nonhydrostatic_simulation.jl:43]: https://github.com/tkluck/StatProfilerHTML.jl

Comment (hennyg888) [reply benchmark/profiliable_nonhydrostatic_simulation.jl:43]: Neat! That's definitely a very nice way to output interactive flame graphs. I'll give it a try for this script and something for  #1919 and #1923. We might be able to embed or link the whole output HTML file directly onto github comments.

**Comment by glwagner** [comment]
I think we should close this PR since the tests are failing. We can easily reproduce this script in the future if we need to.

**Pull request closed without merging**